src/fragments/lib/datastore/js/data-access/observe-update-snippet.mdx

```diff
@@ -0,0 +1,55 @@
+```ts
+// Example showing how to observe the model and keep state updated before
+// performing a save. This uses the useEffect React hook, but you can
+// substitute for a similar mechanism in your application lifecycle with
+// other frameworks.
```

Comment:
This seems like a very complex example on this page. we're also using `observeQuery` before the "Real Time" section, but understandably, the paragraph before this example does reference that section. 

I'm thinking what the ideal would be... 
- "Getting started" section 1 remains the same
- "Manipulating data" section 2, under "**Avoid working with stale data!** points to section 4
- "Real time" - move this up from section 7 to section 3
- "Putting it all together" - section 4, talk about the two options - "observe and save" and "query then save" . When talking about "observe and save", reference back to section 3 "real time" and then show the `observe-update-snippet` example. then a brief explanation that if you don't subscribe to data and update the models in memory, then you'll have to perform the "query then save" option. This is so that we provide the recommended way is "observe and save" from the indepth example code here, but explain if they don't subscribe to data, they should query-merge-save their models.

Reply:
Yeah, the linearity issue makes sense. But, it also seems a bit out of place to put a "Putting it all together" link in the middle of the list. It's more like, "putting the things we've introduced so far together." 😛 Having a discussion about new pages and menu ordering makes me concerned that we'll rabbit hole on information architecture, whereas I think it's more pressing to get the example code out in a _reasonable_ spot soon.

That said, I could add the examples to the realtime data page instead in a new, third page section and just provide a link here. It may improve linearity. And, it doesn't _seem_ like an IA rabbit hole to me.

What do you think?

(Then again, maybe I'm already *in* an IA rabbit hole! 🐇 ⛳)

Reply:
I agree we shouldn't rabbit hole on where the information is, better to decouple that from the example itself, and also placing it on the realtime data page may improve linearity, so i'll leave it for you to decide, either keep it here and let it sit in production to get feedback, or attempt the move to realtime data page now